addon/src/-private/array.ts

```diff
@@ -106,6 +106,18 @@ class TrackedArray<T = unknown> {
       get(target, prop /*, _receiver */) {
         let index = convertToInt(prop);
 
+        if (prop === 'splice') {
+          return (
+            ...args: [start: number, deleteCount?: number | undefined]
+          ) => {
+            try {
+              return target.splice(...args);
```

Comment:
Your solution here @lifeart was about creating new `TrackedArray`, but [this test](https://github.com/tracked-tools/tracked-built-ins/blob/master/test-app/tests/unit/array-test.ts#L338) explicitly asks about it _not_ being TrackedArray.

1) Is that still correct assumption?
2) If yes, should we instead add a test that proves it's native `Array`? What it's _not_ seem like a weak test to me.

Reply:
@MichalBryxi it's just my guess, because I'm assume if we use `TrackedArray`, all it's mutations should return `TrackedArrays` too, and splice return `part` of array, and IMHO it should be tracked.

But it's likely a question to repo maintainers, cc @NullVoxPopuli

Reply:
My guess would be that we don't want things that were not previously explicitly marked as tracked being tracked. Not sure if this case if valid, but I guess it's about preventing those:

```
@tracked myArray = ['foo', 'bar'];
someOtherVar;

someOtherVar = myArray.splice(0, 1); // now someOtherVar is reactive/tracked, wut?
```

Maybe?

Reply:
I think proposed sample is not a case, because with `@tracked` decorator we making only property reactive.
Use-case aligned with this lib looks like:

```js
const myArray = trakced(['foo','bar']);
const trackedResult = myArray.splice(0,1);
```

Reply:
Yes, sorry, that's what I meant. Just morning brain ...